blocks/subscriptions-block/features/checkout/_children/PaymentInfo/index.jsx

```diff
@@ -1,51 +1,174 @@
-import React, { useEffect, useState } from "react";
+import React, { useEffect, useState, useMemo } from "react";
 import { Elements } from "@stripe/react-stripe-js";
 import { loadStripe } from "@stripe/stripe-js";
+import useSales from "../../../../components/useSales";
+import PayPal from "../../../../components/PayPal";
 
-import { usePhrases } from "@wpmedia/arc-themes-components";
+import { usePhrases, Heading, HeadingSection, Paragraph } from "@wpmedia/arc-themes-components";
 
 import PaymentForm from "../../../../components/PaymentForm";
 
-const PaymentInfo = ({ orderNumber, paymentDetails, paymentMethodID, successURL, className }) => {
+const PaymentInfo = ({
+	successURL,
+	className,
+	userInfo,
+	offerURL,
+	paypal,
+	stripeIntents,
+	errorPaymentOption,
+	isInitialized,
+	LABEL_ORDER_NUMBER_PAYPAL
+}) => {
+	const { Sales } = useSales();
+	window.Sales = Sales;
 	const [stripeInstance, setStripeInstance] = useState(null);
 
-	// initialized payment doc https://redirector.arcpublishing.com/alc/en/arc-xp-subscriptions-sdks?id=kb_article_view&sys_kb_id=7770d58747447990eee38788436d4362&spa=1
-	const { parameter2: stripeKey, parameter1: clientSecret } = paymentDetails;
+	const [isStripeInitialized, setIsStripeInitialized] = useState(false);
+	const [isPayPal, setIsPayPal] = useState(false);
+
+	const [orderNumber, setOrderNumber] = useState();
+	const [orderNumberPayPal, setOrderNumberPayPal] = useState();
+	const [payment, setPayment] = useState({});
+
+	// load stripe key via payment details stripe key string
+	const { parameter2: stripeKey, parameter1: clientSecret } = payment;
 
 	const phrases = usePhrases();
 
 	const formErrorText = phrases.t("subscriptions-block.payment-error");
 	const formLabel = phrases.t("subscriptions-block.credit-card-information");
-	const formTitle = phrases.t("subscriptions-block.payment-information");
 	const submitText = phrases.t("subscriptions-block.submit-payment");
+	const payWithCardDividerLabel = phrases.t("subscriptions-block.payWithCard-label");
+
+	useEffect(() => {
+		if (stripeIntents?.paymentMethodID && !errorPaymentOption) {
```

Comment:
I think here we need to add `!isInitialized`.
` if (stripeIntents?.paymentMethodID && !errorPaymentOption && !isInitialized)`, in order to avoid to call get and create a new cart, if we are trying to call finalize